common/unsafe/src/main/java/org/apache/spark/unsafe/Platform.java

```diff
@@ -37,6 +38,32 @@
 
   public static final int DOUBLE_ARRAY_OFFSET;
 
+  private static final boolean unaligned;
+  static {
+    boolean _unaligned;
+    // use reflection to access unaligned field
+    try {
+      Class<?> bitsClass = Class.forName("java.nio.Bits", false,ClassLoader.getSystemClassLoader());
+      Method unalignedMethod = bitsClass.getDeclaredMethod("unaligned");
+      unalignedMethod.setAccessible(true);
+      _unaligned = Boolean.TRUE.equals(unalignedMethod.invoke(null));
+    } catch (Throwable t) {
+      // We at least know x86 and x64 support unaligned access.
+      String arch = SystemPropertyUtil.get("os.arch", "");
```

Comment:
SystemPropertyUtil is a netty class. It's fine to just use the Java `System.getProperty`.